cognite/client/data_classes/contextualization.py

```diff
@@ -1036,3 +1038,212 @@ def __init__(
+
+        If you want to use an undocumented parameter, you can pass it as a keyword argument:
+
+            >>> config_with_undoducmented_params = DiagramDetectConfig(
+            ...     remove_leading_zeros=True,
+            ...     connection_flags=ConnectionFlags(new_undocumented_flag=True),
+            ...     new_undocumented_param={"can_be_anything": True},
+            ... )
+    """
+
+    def __init__(
+        self,
+        annotation_extract: bool | None = None,
+        case_sensitive: bool | None = None,
+        connection_flags: ConnectionFlags | list[str] | None = None,
+        customize_fuzziness: CustomizeFuzziness | dict[str, Any] | None = None,
+        direction_delta: float | None = None,
+        direction_weights: DirectionWeights | dict[str, Any] | None = None,
+        min_fuzzy_score: float | None = None,
+        read_embedded_text: bool | None = None,
+        remove_leading_zeros: bool | None = None,
+        substitutions: dict[str, list[str]] | None = None,
+        **params: Any,
+    ) -> None:
+        self.annotation_extract = annotation_extract
+        self.case_sensitive = case_sensitive
+        self.connection_flags = connection_flags
+        self.customize_fuzziness = customize_fuzziness
+        self.direction_delta = direction_delta
+        self.direction_weights = direction_weights
+        self.min_fuzzy_score = min_fuzzy_score
+        self.read_embedded_text = read_embedded_text
+        self.remove_leading_zeros = remove_leading_zeros
+        self.substitutions = substitutions
+
+        _known_params = {to_camel_case(k): k for k in vars(self)}
+        self._extra_params = {}
+        for param_name, value in params.items():
+            if known := _known_params.get(to_camel_case(param_name)):
+                raise ValueError(
+                    f"Provided parameter name `{param_name}` collides with a known parameter `{known}`. Please use it insead."
+                )
+            self._extra_params[param_name] = value
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        dumped = super().dump(camel_case=camel_case)
+
+        for param_name, v in self._extra_params.items():
+            dumped[to_camel_case(param_name) if camel_case else param_name] = v
+
+        remove_keys: list[str] = []
+        for param_name, v in dumped.items():
+            if isinstance(v, CogniteObject):
+                dumped[param_name] = v.dump(camel_case=camel_case)
+            elif isinstance(v, ConnectionFlags):
+                cf = v.dump()
+                if cf:
+                    dumped[param_name] = cf
+                else:
+                    remove_keys.append(param_name)
+
+        for param_name in remove_keys:
+            del dumped[param_name]
+
+        return dumped
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        resource_copy = resource.copy()
```

Comment:
@haakonvt I do the copy and pop because I expect undocumented parameters to be in this resource dict.

If I use get and then pass the original dict into kwargs, the parsed and loaded params like `ConnectionFlags` will be passed to `cls()` twice, causing a TypeError.

I need some way to separate known loadable params from unknown ones, so I decided to use `pop()` to remove the items we need to load separately.